ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -1182,6 +1188,26 @@ void Matmul::validate(
             // TODO: For 1D and 2D mcasts, we don't check if tensor is single core or single row/col
             // We can uplift these variants to skip mcasting to support single core (1D) or single row/col (2D)
             if constexpr (std::is_same_v<ProgramConfigType, MatmulMultiCoreReuseMultiCast1DProgramConfig>) {
+                TT_FATAL(
```

Comment:
should we add some constrains here in case that output is sharded?
In general more descriptives error messges which would say which variables have to divisble with some other variable would be helpful